src/Illuminate/Database/Eloquent/Relations/Concerns/InteractsWithPivotTable.php

```diff
@@ -349,7 +349,7 @@ public function detach($ids = null, $touch = true)
         // associations, otherwise all of the association ties will be broken.
         // We'll return the numbers of affected rows when we do the deletes.
         if (! is_null($ids = $this->parseIds($ids))) {
-            if (count($ids) === 0) {
+            if (count((array) $ids) === 0) {
```

Comment:
I'm confused. Why does the `null` check prior to this line not stop using getting here in the first place?